homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,286 @@
+    CONNECTION,
+    DOMAIN as AGENT_DOMAIN,
+    SERVICE_UPDATE,
+)
+from .helpers import service_signal
+
+SCAN_INTERVAL = timedelta(seconds=CAMERA_SCAN_INTERVAL_SECS)
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEV_EN_ALT = "enable_alerts"
+_DEV_DS_ALT = "disable_alerts"
+_DEV_EN_REC = "start_recording"
+_DEV_DS_REC = "stop_recording"
+_DEV_SNAP = "snapshot"
+
+CAMERA_SERVICES = {
+    _DEV_EN_ALT: "async_enable_alerts",
+    _DEV_DS_ALT: "async_disable_alerts",
+    _DEV_EN_REC: "async_start_recording",
+    _DEV_DS_REC: "async_stop_recording",
+    _DEV_SNAP: "async_snapshot",
+}
+
+
+async def async_setup_entry(
+    hass, config_entry, async_add_entities, discovery_info=None
+):
+    """Set up the Agent cameras."""
+    filter_urllib3_logging()
+    cameras = []
+
+    server = hass.data[AGENT_DOMAIN][config_entry.entry_id][CONNECTION]
+    server.cameras = []
+    if not server.deviceList:
+        _LOGGER.warning("Could not fetch cameras from Agent server")
+        return
+
+    for device in server.deviceList:
+        if device.typeID == 2:
+            camera = AgentCamera(device)
+            cameras.append(camera)
+            server.cameras.append(camera)
+
+    async_add_entities(cameras)
+
+    platform = entity_platform.current_platform.get()
+    for service, method in CAMERA_SERVICES.items():
+        platform.async_register_entity_service(
+            service, {vol.Required("entity_id"): str}, method
+        )
+
+    return True
+
+
+class AgentCamera(MjpegCamera):
+    """Representation of an Agent Device Stream."""
+
+    def __init__(self, device, enabled_default: bool = True):
+        """Initialize as a subclass of MjpegCamera."""
+        self._servername = device.client.name
+        self.server_url = device.client._server_url
+
+        device_info = {
+            CONF_NAME: device.name,
+            CONF_MJPEG_URL: f"{self.server_url}{device.mjpeg_image_url}&size=640x480",
+            CONF_STILL_IMAGE_URL: f"{self.server_url}{device.still_image_url}&size=640x480",
+        }
+        self._unsub_dispatcher = []
+        self.device = device
+        self._should_poll = True
+        self._removed = False
+        self._name = f"{self._servername} {device.name}"
+        self._stream_url = f"{self.server_url}{device.mp4_url}"
+        self._unique_id = f"{device._client.unique}_{device.typeID}_{device.id}"
+        self._enabled_default = enabled_default
+        super().__init__(device_info)
+
+    @property
+    def device_info(self):
+        """Return the device info for adding the entity to the agent object."""
+        return {
+            "identifiers": {(AGENT_DOMAIN, self._unique_id)},
+            "name": self._name,
+            "manufacturer": "Agent",
+            "model": "Camera",
+            "sw_version": self.device.client.version,
+        }
+
+    @property
+    def should_poll(self) -> bool:
+        """Update the state periodically."""
+        return self._should_poll
+
```

Comment:
```suggestion
```

should_poll is the default. Since you always set this to true, you can delete this.